java/src/test/java/ai/onnxruntime/OnnxTensorTest.java

```diff
@@ -87,6 +87,83 @@ public void testScalarCreation() throws OrtException {
     }
   }
 
+  @Test
+  public void testBufferCreation() throws OrtException {
+    OrtEnvironment env = OrtEnvironment.getEnvironment();
+
+    float[] arrValues = new float[] {0, 1, 2, 3, 4};
+    try (OnnxTensor t = OnnxTensor.createTensor(env, arrValues)) {
+      // array creation isn't backed by buffers
+      Assertions.assertFalse(t.isCopy());
```

Comment:
After looking at the test code, it seems that `isCopy()` needs to be renamed to `isBufferBacked()` and then make a good comment in the class, explaining what that means and its ramifications.

Reply:
That doesn't fully capture what `isCopy` currently means, but I agree it is confusing. There are three ways of creating an `OnnxTensor` in Java, by passing in a multidimensional array, by passing in a non-direct byte buffer, and by passing in a direct byte buffer. `isBufferBacked` wouldn't distinguish between the two byte buffer based methods, yet in the first byte buffer case the user can't modify their reference to the original buffer to change the OrtValue, but in the second case they can. The array vs buffer case is distinguished by `getBufferRef` returning an empty optional vs a full optional.

Once this is merged I plan to migrate the array case over so that it allocates a direct byte buffer and copies the array to it in Java before passing the buffer to ORT. This will a) simplify the JNI code massively as the array traversal code is nasty in C, b) make it faster as there will be fewer C <-> JVM transitions, and c) mean that `isCopy` only has a single meaning which is "is the backing byte buffer a copy of some other memory". That's also a fairly big and ugly change (especially if I modify the `getValue` call to do the array creation in Java too, which is probably more efficient for similar reasons), so it made sense to me to do this smaller change first.